### PR TITLE
Showcase Page

### PR DESCRIPTION
**Pull request by CalvinNolan**
A new page on the website to be a hub for our Technology Showcases.
It shows a number of Showcases each with a small section on every item shown at the showcase.
All this data can be input through the admin console.

### CODE REVIEW

**Comment by 22a** [comment]
What's brad pitt still doing in here?

**Comment by CalvinNolan** [comment]
@22a It's very important that we have realistic tests

File: app/assets/stylesheets/showcase.scss [hunk 1]
```diff
@@ -0,0 +1,232 @@
+.showcase-page {
```
Comment (houli) [inline app/assets/stylesheets/showcase.scss:1]: This whole file should be 2-space indented

File: app/views/showcase/_showcase.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<% if(active) %>
```
Comment (houli) [inline app/views/showcase/_showcase.html.erb]: No need for parens

File: app/views/showcase/_showcase.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+<li class="accordion-navigation active showcase">
+  <a href="#showcase-panel<%= showcase.id %>" class="showcase-header"><%= showcase.date.strftime "%-d %B %Y" %></a>
+  <div id="showcase-panel<%= showcase.id %>" class="content active">
+<% else %>
+<li class="accordion-navigation showcase">
+  <a href="#showcase-panel<%= showcase.id %>" class="showcase-header"><%= showcase.date.strftime "%-d %B %Y" %></a>
+  <div id="showcase-panel<%= showcase.id %>" class="content">
+<% end %>
+    <% showcase.showcase_items.each_with_index do |item, index| %>
+    	<% if(index%2 == 0) %>
```
Comment (houli) [inline app/views/showcase/_showcase.html.erb]: No need for parens. Also using `index.even?` would probably make more sense

File: app/views/showcase/_showcase_item.html.erb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<% if(left) %>
```
Comment (houli) [inline app/views/showcase/_showcase_item.html.erb]: No need for parens

File: app/views/showcase/_showcase_item.html.erb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+<div class="showcase-item left-item">
```
Comment (houli) [inline app/views/showcase/_showcase_item.html.erb]: There are utility classes for left and right in Foundation. Do we have to write our own?

File: app/views/showcase/_showcase_item.html.erb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+<% else %>
+<div class="showcase-item right-item">
+<% end %>
+  <div class="item-content">
+    <h3>
+      <%= item.title %>
+    </h3>
+    <div class="meta">
+      <p class="owner"><%= item.owner %>
+      <%= link_to truncate(item.link[11..item.link.length], :length => 25, :omission => '...'), item.link %></p>
```
Comment (houli) [inline app/views/showcase/_showcase_item.html.erb]: I don't get it. What's this magic 11 for?

Comment (houli) [inline app/views/showcase/_showcase_item.html.erb]: Also might as well use the nicer hash syntax `length: 25, omission: '...'`

File: app/views/showcase/_showcase_item.html.erb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    </div>
+    <div class="text">
+        <%= item.description.html_safe.truncate_words(40) %>
+    </div>
+  </div>
+  <% if(!left) %>
```
Comment (houli) [inline app/views/showcase/_showcase_item.html.erb]: PARENS! :smile:

File: app/views/showcase/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    <h2><span class="orange-text">&lt;</span>ducss<span class="orange-text">/&gt;</span></h2>
+    <h3>Showcase</h3>
+  </div>
+
+  <div class="info">
+    <p class="blurb">We all have great ideas for nifty side projects. In DUCSS we want to create a community where creators and tech enthusiasts can support and inspire one another. Our Technology Showcase events are held once a month for all our members to show off their work and ideas.  This page serves as a hub for our showcases and everyone that presents at them, so come and get a spark of inspiration or find out where to start!</p>
+    <p class="slack"><%= image_tag "slack_icon.png" %> Send us a message to join our slack team <a id="contact-button" href="#" data-reveal-id="contact"><span class="orange-text">DUCSSShowcase</span></a>!</p>
+  </div>
+  <% if @next_showcase %>
+  <div class="small-12 columns services-header">Our Next Showcase</div>
```
Comment (houli) [inline app/views/showcase/index.html.erb]: Indent under the if

File: app/views/showcase/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+      <a class="showcase-header" style="cursor: default"><%= @next_showcase.date.strftime "%-d %B %Y %H:%M" %></a>
+    </li> 
+  </ul>
+  <% end %>
+
+  <div class="small-12 columns services-header">Previous Showcases</div>
+  <hr>
+  <ul class="accordion" data-accordion>
+    <% @previous_showcases.each_with_index do |showcase, index| %>
+      <% if(index == 0) %>
```
Comment (houli) [inline app/views/showcase/index.html.erb]: ()

File: spec/controllers/showcase_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe ShowcaseController, type: :controller do
+	describe 'GET #index' do
```
Comment (houli) [inline spec/controllers/showcase_controller_spec.rb]: Indentation

File: spec/factories/showcase_items.rb [hunk 11]
```diff
@@ -0,0 +1,10 @@
+FactoryGirl.define do
+  factory :showcase_item do
+    owner "Brad Pitt"
+		link "http://www.github.com/BradPitt"
```
Comment (houli) [inline spec/factories/showcase_items.rb]: Indentation

File: spec/helpers/application_helper_spec.rb [hunk 12]
```diff
@@ -26,5 +26,12 @@
         expect(helper.full_title(title)).to eq "#{title} - #{base}"
       end
     end
+
+    context 'if title is not empty' do
```
Comment (houli) [inline spec/helpers/application_helper_spec.rb]: Don't need this test. Already covered by the above

File: app/views/showcase/_showcase_item.html.erb (hunk 6 above)
Comment (CalvinNolan) [inline app/views/showcase/_showcase_item.html.erb]: it omits the 'http://www.' part of the url

Comment (houli) [inline app/views/showcase/_showcase_item.html.erb]: I think a more robust solution would be to use URI.parse

File: app/views/showcase/index.html.erb (hunk 9 above)
Comment (houli) [inline app/views/showcase/index.html.erb]: Also `.even?`

Comment (CalvinNolan) [inline app/views/showcase/index.html.erb]: For where?

File: app/views/showcase/_showcase.html.erb [hunk 13]
```diff
@@ -0,0 +1,18 @@
+<% if active %>
+<li class="accordion-navigation active showcase">
+  <a href="#showcase-panel<%= showcase.id %>" class="showcase-header"><%= showcase.date.strftime "%-d %B %Y" %></a>
+  <div id="showcase-panel<%= showcase.id %>" class="content active">
+<% else %>
+<li class="accordion-navigation showcase">
+  <a href="#showcase-panel<%= showcase.id %>" class="showcase-header"><%= showcase.date.strftime "%-d %B %Y" %></a>
+  <div id="showcase-panel<%= showcase.id %>" class="content">
+<% end %>
+    <% showcase.showcase_items.each_with_index do |item, index| %>
```
Comment (houli) [inline app/views/showcase/_showcase.html.erb:10]: Didn't notice the first time but the indentation here

File: app/views/showcase/index.html.erb [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+        <a class="showcase-header" style="cursor: default"><%= @next_showcase.date.strftime "%-d %B %Y %H:%M" %></a>
+      </li> 
+    </ul>
+  <% end %>
+
+  <div class="small-12 columns services-header">Previous Showcases</div>
+  <hr>
+  <ul class="accordion" data-accordion>
+    <% @previous_showcases.each_with_index do |showcase, index| %>
+      <% if index == 0 %>
```
Comment (houli) [inline app/views/showcase/index.html.erb]: even?

Comment (CalvinNolan) [inline app/views/showcase/index.html.erb]: w h e r e ?

Comment (houli) [inline app/views/showcase/index.html.erb]: `index == 0`

Comment (CalvinNolan) [inline app/views/showcase/index.html.erb]: That's to open the first accordion. We don't want every even accordion open.

Comment (houli) [inline app/views/showcase/index.html.erb]: Ah sorry, I think in that case we might just go forgo the each_with_index and index variables and see `if showcase == showcases.first`

**Pull request merged**